### ISSUE TITLE
convert(::Type{SubString{S}}, , s::AbstractString) broken?

### ISSUE THREAD

**Issue by samoconnor** [JuliaLang/julia#25758]
See https://github.com/JuliaLang/julia/commit/5167f17355180047ef27eccfea1d5f0e635843e5#r26877190

```julia
$ ~/julia/julia/julia -e 'println(VERSION);
struct foo bar::SubString{String} end ;
x = foo(SubString("foo", 1, 0)) ;
@show x.bar.string'
0.7.0-DEV.3429
(x.bar).string = ""

$ julia -e 'println(VERSION);
struct foo bar::SubString{String} end ;
x = foo(SubString("foo", 1, 0)) ;
@show x.bar.string'
0.6.2
x.bar.string = "foo"
```

**Comment by samoconnor** [comment]
Need this?
```julia
Base.convert(::Type{SubString{S}}, s::SubString{S}) where {S<:AbstractString} = s
```

**Comment by nalimilan** [comment]
Cc: @bkamins

**Comment by nalimilan** [comment]
I guess that's https://github.com/JuliaLang/julia/pull/25525.

**Comment by bkamins** [comment]
Seems so - running the code with that PR prints `"foo"`.

**Comment by JeffBezanson** [comment]
Fixed by #25525.
